autoconfigure/collector-sqs/src/test/java/zipkin/autoconfigure/collector/sqs/ZipkinSqsCollectorAutoConfigurationTest.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2016 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.autoconfigure.collector.sqs;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.boot.autoconfigure.PropertyPlaceholderAutoConfiguration;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import zipkin.collector.CollectorMetrics;
+import zipkin.collector.CollectorSampler;
+import zipkin.collector.sqs.AwsSqsCollector;
+import zipkin.storage.InMemoryStorage;
+import zipkin.storage.StorageComponent;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import static org.springframework.boot.test.util.EnvironmentTestUtils.addEnvironment;
+
+public class ZipkinSqsCollectorAutoConfigurationTest {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  AnnotationConfigApplicationContext context;
+
+  @After
+  public void close() {
+    if (context != null) context.close();
+  }
+
+  @Test
+  public void doesntProvideCollectorComponent_whenSqsQueueUrlUnset() {
+    context = new AnnotationConfigApplicationContext();
+    context.register(PropertyPlaceholderAutoConfiguration.class,
+        ZipkinSqsCollectorAutoConfiguration.class, InMemoryConfiguration.class);
+    context.refresh();
+
+    thrown.expect(NoSuchBeanDefinitionException.class);
+    context.getBean(AwsSqsCollector.class);
+  }
+
+  @Test
+  public void provideCollectorComponent_whenSqsQueueUrlIsSet() {
```

Comment:
so to check the behavior that @shakuzen was curious about (if setters are needed in property types), we could replicate this test saying like 'setsZipkinSqsCollectorProperties()' and in there ..

```java
assertThat(context.getBean(ZipkinSqsCollectorProperties.class).getQueueUrl())
  .isEqualTo("http://localhost:1234");
```